src/context/innerContext/InnerContext.ts

```diff
@@ -1,14 +1,31 @@
 import React from 'react';
 import {ItemMap} from '../../containers/PageConstructor/PageConstructor';
-import {LoadableConfig} from '../../models';
+import {ConstructorItem, LoadableConfig} from '../../models';
+
+export type ItemWrapper = (
+    item: React.ReactElement,
+    key: string,
+    data: ConstructorItem,
+    context: WrapperContext,
+) => React.ReactElement;
 
 export interface InnerContextType {
     blockTypes: string[];
+    subBlockTypes: string[];
+    headerBlockTypes: string[];
     itemMap: ItemMap;
     loadables?: LoadableConfig;
+
+    itemWrappers: ItemWrapper[];
 }
 
+export type WrapperContext = Omit<InnerContextType, 'itemWrappers'>;
+
 export const InnerContext = React.createContext<InnerContextType>({
     blockTypes: [],
+    subBlockTypes: [],
+    headerBlockTypes: [],
+
```

Comment:
why empty line?

Reply:
To separate blocks from other elements. Will remove.

Reply:
Sorry, now we don't use this rule. If you think that it is better, you can leave it :)